torch/_meta_registrations.py

```diff
@@ -1398,13 +1398,50 @@ def meta__fused_adam_(
     grad_scale=None,
     found_inf=None,
 ):
-    for l in [self, grads, exp_avgs, exp_avg_sqs, max_exp_avg_sqs, state_steps]:
+    for l in [params, grads, exp_avgs, exp_avg_sqs, max_exp_avg_sqs, state_steps]:
         check(
-            isinstance(self, List),
-            lambda: f"exponent must be a tensor list but got {type(self)}",
+            isinstance(l, List),
+            lambda: f"exponent must be a tensor list but got {type(l)}",
         )
 
 
+@register_meta([aten._fused_adam.default])
```

Comment:
does this address the issue you encountered?

Reply:
Yep! resolved the most part, thanks for the suggestion! I think there're still some issues with regard to fake tensor fallback "val" set, documented in https://github.com/pytorch/pytorch/issues/99356